prime-router/metadata/fhir_mapping/hl7/secondary/Source.yml

```diff
@@ -16,18 +43,57 @@ endpoint_1:
   constants:
     oidUrlPrefix: "urn:oid:"
 
-# For all other ID types
+# For UUIDs
 endpoint_2:
   type: URL
-  condition: $idStr NOT_NULL && $systemStr NOT_EQUALS ISO
+  condition: $idStr NOT_NULL && $systemStr EQUALS UUID
   valueOf: $value
   expressionType: HL7Spec
   vars:
     idStr: String, MSH.3.2
-    value: $idUrlPrefix + MSH.3.2
+    value: $oidUrlPrefix + MSH.3.2
     systemStr: String, MSH.3.3
   constants:
-    idUrlPrefix: "urn:id:"
+    oidUrlPrefix: "urn:uuid:"
+
+# For DNS
+endpoint_3:
+  type: URL
+  condition: $idStr NOT_NULL && $systemStr EQUALS DNS
+  valueOf: $value
+  expressionType: HL7Spec
+  vars:
+    idStr: String, MSH.3.2
+    value: $oidUrlPrefix + MSH.3.2
+    systemStr: String, MSH.3.3
+  constants:
+    oidUrlPrefix: "urn:dns:"
+
+# For CLIA
+endpoint_4:
+  type: URL
+  condition: $idStr NOT_NULL && $systemStr EQUALS CLIA
+  valueOf: $value
+  expressionType: HL7Spec
+  vars:
+    idStr: String, MSH.3.2
+    value: $oidUrlPrefix + MSH.3.2
+    systemStr: String, MSH.3.3
+  constants:
+    oidUrlPrefix: "urn:clia:"
+
+# For URIs
+endpoint_5:
+  type: URL
+  condition: $idStr NOT_NULL && $systemStr EQUALS URI
```

Comment:
Note: Something else to keep in mind is that EQUALS only works here if `systemStr` is set as string.  E.g. it will not work is you just do `systemStr: MSH.3.3` <- missing `String, `

Reply:
This should also be documented somewhere in a file or doc

Reply:
Will create an issue to go into the Linux4Heatlh repo and submit a PR with documentation changes.

Reply:
We talked about it, and a ticket will be written to address these later, focusing on the library itself